File: OpenHome/Net/Bindings/Java/org/openhome/net/controlpoint/tests/TestCpDeviceDv.java
package org.openhome.net.controlpoint.tests;

import java.net.Inet4Address;


import org.openhome.net.controlpoint.CpDeviceDv;
import org.openhome.net.core.InitParams;
import org.openhome.net.core.Library;
import org.openhome.net.core.NetworkAdapter;
import org.openhome.net.core.SubnetList;
import org.openhome.net.device.tests.DeviceBasic;

public class TestCpDeviceDv
{
	
	public static void main(String[] args)
	{
		InitParams initParams = new InitParams();
		initParams.setMsearchTimeSecs(1);
		initParams.setUseLoopbackNetworkAdapter();
		Library lib = new Library();
		lib.initialise(initParams);
		SubnetList subnetList = new SubnetList();
		NetworkAdapter nif = subnetList.getSubnet(0);
		Inet4Address subnet = nif.getSubnet();
		subnetList.destroy();
		lib.startCombined(subnet);
		
		System.out.println("TestCpDeviceDvJava - starting");
		DeviceBasic device = new DeviceBasic();
		CpDeviceDv cpDevice = new CpDeviceDv(device.getDevice());
		TestBasicCp cp = new TestBasicCp(cpDevice);
		cp.testActions();
		cp.testSubscriptions();
		cpDevice.removeRef();
		device.dispose();
		System.out.println("TestCpDeviceDvJava - completed");
	}

}

File: OpenHome/Net/Bindings/Cs/ControlPoint/Tests/TestCpDeviceDv.cs
using System;
using System.Collections.Generic;
using System.Text;
using System.Threading;
using System.Diagnostics;
using OpenHome.Net.Core;
using OpenHome.Net.ControlPoint;
using OpenHome.Net.Device;

namespace OpenHome.Net
{
    class Program
    {
        public static void Main(string[] args)
        {
            InitParams initParams = new InitParams
            {
                MsearchTimeSecs = 1,
                UseLoopbackNetworkAdapter = true
            };
            using (Library lib = Library.Create(initParams))
            {
                SubnetList subnetList = new SubnetList();
                NetworkAdapter nif = subnetList.SubnetAt(0);
                uint subnet = nif.Subnet();
                subnetList.Dispose();
                lib.StartCombined(subnet);

                Console.Write("TestCpDeviceDvCs - starting\n");
                DeviceBasic device = new DeviceBasic();
                CpDeviceDv cpDevice = new CpDeviceDv(device.Device());
                TestBasicCp cp = new TestBasicCp(cpDevice);
                cp.TestActions();
                cp.TestSubscriptions();
                cpDevice.RemoveRef();
                device.Dispose();
                Console.Write("TestCpDeviceDvCs - completed\n");
            }
        }
    }
}
